vstudio/wnbdevents.xml

```diff
@@ -0,0 +1,72 @@
+﻿<?xml version='1.0' encoding='utf-8' standalone='yes'?>
+<instrumentationManifest
+    xmlns="http://schemas.microsoft.com/win/2004/08/events"
+    xmlns:win="http://manifests.microsoft.com/win/2004/08/windows/events"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://schemas.microsoft.com/win/2004/08/events eventman.xsd">
+    <instrumentation>
+        <events>
+            <provider
+                guid="{FFACC4E7-C115-4FE2-9D3C-80FAE73BAB91}"
+                messageFileName="%WINDIR%\System32\drivers\wnbd.sys"
+                name="WNBD"
+                resourceFileName="%WINDIR%\System32\drivers\wnbd.sys"
+                symbol="WNBD">
+                <channels>
+                    <importChannel
+                        chid="SYSTEM"
+                        name="System"/>
+                </channels>
+                <templates>
+                    <template tid="tid_load_template">
+                        <data name="FunctionName"
+                              inType="win:AnsiString"
+                              outType="xs:string"/>
+                        <data name="LineNumber"
+                              inType="win:UInt32"
+                              outType="xs:unsignedInt"/>
+                        <data name="Message"
+                              inType="win:AnsiString"
+                              outType="xs:string"/>
+                    </template>
+                </templates>
+                <events>
+                    <event
+                        channel="SYSTEM"
+                        level="win:Error"
+                        message="$(string.ErrorEvent.EventMessage)"
+                        opcode="win:Info"
```

Comment:
Please correct me if I'm wrong, but from what I can tell ``opcode`` is only used Windows versions older than Windows 7. Can we skip it? It's a bit confusing that we have ``opcode="win:Info"`` and ``level="win:Error".

Reply:
Documentation say it started since Vista: https://docs.microsoft.com/en-us/windows/win32/wes/eventmanifestschema-opcodetype-complextype .

The element is not mandatory and it is specified to differentiate between events which should be logged directly to the system and the ones which are logged on demand via utilities (i.e. `tracelog`).

AFAIK the default would be zero (`win:Info`) but in my opinion that would be even more confusing because that would require even more additional knowledge.